model/event_models.go

```diff
@@ -0,0 +1,104 @@
+package model
+
+import (
+	"time"
+
+	"gorm.io/datatypes"
+)
+
+type PauseEvent struct {
+	PausedAt *string `json:"paused_at"`
+}
+
+type AvailabilityStatusEvent struct {
+	AvailabilityStatus *string `json:"availability_status"`
+	LastCheckedAt      *string `json:"last_checked_at"`
+	LastAvailableAt    *string `json:"last_available_at"`
+}
+
+type ApplicationEvent struct {
+	AvailabilityStatusEvent
+	PauseEvent
+
+	ID        int64   `json:"id"`
+	CreatedAt *string `json:"created_at"`
+	UpdatedAt *string `json:"updated_at"`
+
+	AvailabilityStatusError *string        `json:"availability_status_error"`
+	Extra                   datatypes.JSON `json:"extra"`
+	SuperkeyData            datatypes.JSON `json:"superkey_data"`
+
+	SourceID          int64   `json:"source_id"`
+	ApplicationTypeID int64   `json:"application_type_id"`
+	Tenant            *string `json:"tenant"`
+}
+
+type AuthenticationEvent struct {
+	ID        string    `json:"id"`
+	CreatedAt time.Time `json:"created_at"`
+
+	Name                    string                 `json:"name"`
+	AuthType                string                 `json:"authtype"`
+	Username                string                 `json:"username"`
+	Password                string                 `json:"password"`
```

Comment:
We should remove this field - we don't produce it for the event stream since its globally readable by anyone on the platform.